clang/runtime/dpct-rt/include/dpl_extras/vector.h.inc

```diff
@@ -121,143 +122,142 @@ private:
 
   void _set_capacity_and_alloc() {
     _capacity = ::std::max(_size * 2, _min_capacity());
-    _storage = _alloc.allocate(_capacity);
+    _storage = ::std::allocator_traits<Allocator>::allocate(_alloc, _capacity);
+  }
+
+  void _construct(size_type n, size_type start_idx = 0) {
+    if (n > 0) {
+      for (int i = 0; i < n; i++) {
```

Comment:
I'm not sure we can assume the thread safety of construct or destroy calls.  

If we could assume this thread safety, from a technical perspective we cant send these construct calls to a SYCL kernel because of the use of variadic functions.  However, if we could be sure that the construction are thread safe we could either use a `par_unseq` policy or create our own version of `std::allocator_traits` with a fixed number of possible arguments to the construct call to avoid the variadic function call allow us to use them in SYCL kernels.  I have a prototype of something like this from a different implementation of this code.